src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -72,4 +73,11 @@ default Logger getLogger() {
      */
     Optional<Object> getInstance();
 
+    /**
+     * Returns the plugin's internal service manager.
+     *
+     * @return Internal service manager
+     */
+    ServiceManager getServiceManager();
```

Comment:
Please read the PR description. Also I would appreciate it if you could refrain from making formatting / grammar comments at this time. As I said this is still very much a WIP and the groundwork is just being laid.